src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/MultiSelectList.cs

```diff
@@ -111,7 +112,12 @@ public MultiSelectList(IEnumerable items, string dataValueField, string dataText
 
         public virtual IEnumerator<SelectListItem> GetEnumerator()
         {
-            return GetListItems().GetEnumerator();
+            if (_selectListItems == null)
+            {
+                _selectListItems = GetListItems();
+            }
+
+            return _selectListItems.GetEnumerator();
         }
 
         internal IList<SelectListItem> GetListItems()
```

Comment:
This may have been `internal` for testing at some point. But nothing references it today. Since it's now less performant than using the enumerator directly, make it `private`.